executor/slow_query.go

```diff
@@ -290,32 +418,57 @@ func (e *slowQueryRetriever) parseSlowLog(ctx context.Context, sctx sessionctx.C
 	defer close(ch)
 	for {
 		startTime := time.Now()
-		log, err := e.getBatchLog(reader, &offset, logNum)
+		var logs [][]string
+		var err error
+		noMoreFiles := false
+		if !e.extractor.Desc {
+			logs, err = e.getBatchLog(reader, &offset, logNum)
+		} else {
+			logs, err = e.getBatchLogForReversedScan(reader, &offset, logNum)
+			file := e.getNextFile()
+			if file != nil {
+				reader.Reset(file)
+			} else {
+				noMoreFiles = true
+			}
```

Comment:
The logic was a little bit strange here. can we move it into `getBatchLogForReversedScan`?